File: app/src/main/java/sugtao4423/calcsalary/DBUtils.java
package sugtao4423.calcsalary;

import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteStatement;

import java.util.ArrayList;

public class DBUtils{

    private SQLiteDatabase db;

    public DBUtils(Context context){
        db = new SQLHelper(context).getWritableDatabase();
    }

    public ArrayList<WorkItem> getWorkItems(){
        ArrayList<WorkItem> result = new ArrayList<>();
        Cursor c = db.rawQuery("SELECT ROWID, * FROM database", null);
        while(c.moveToNext()){
            int rowid = c.getInt(0);
            int hour = c.getInt(1);
            int minute = c.getInt(2);
            String changed = c.getString(3);
            int jikyu = c.getInt(4);
            String memo = c.getString(5);

            WorkItem item = new WorkItem(rowid, jikyu, hour, minute, changed, memo);
            result.add(item);
        }
        c.close();
        return result;
    }

    public void createWorkItem(int hour, int minute, String changed, int jikyu, String memo){
        String sql = "INSERT INTO database VALUES (?, ?, ?, ?, ?)";
        String[] bindArgs = new String[]{
                String.valueOf(hour),
                String.valueOf(minute),
                changed,
                String.valueOf(jikyu),
                memo
        };
        SQLiteStatement stmt = db.compileStatement(sql);
        stmt.bindAllArgsAsStrings(bindArgs);
        stmt.execute();
        stmt.close();
    }

    public void updateWorkTime(int targetRowid, int hour, int minute, String changed, int jikyu, String memo){
        String sql = "UPDATE database SET hour = ?, minute = ?, changed = ?, jikyu = ?, memo = ? WHERE ROWID = ?";
        String[] bindArgs = new String[]{
                String.valueOf(hour),
                String.valueOf(minute),
                changed,
                String.valueOf(jikyu),
                memo,
                String.valueOf(targetRowid)
        };
        SQLiteStatement stmt = db.compileStatement(sql);
        stmt.bindAllArgsAsStrings(bindArgs);
        stmt.execute();
        stmt.close();
    }

    public void deleteWorkItem(int targetRowid){
        String sql = "DELETE FROM database WHERE ROWID = " + targetRowid;
        db.execSQL(sql);
    }

    public void close(){
        db.close();
    }

}

File: app/src/main/java/sugtao4423/calcsalary/DBUtils.kt
package sugtao4423.calcsalary

import android.content.Context

class DBUtils(context: Context) {

    private val db = SQLHelper(context).writableDatabase

    fun getWorkItems(): ArrayList<WorkItem> {
        val result = arrayListOf<WorkItem>()
        db.rawQuery("SELECT ROWID, * FROM salaries", null).apply {
            while (moveToNext()) {
                val rowId = getInt(0)
                val wage = getInt(1)
                val minutes = getInt(2)
                val changed = getString(3)
                val memo = getString(4)

                val item = WorkItem(rowId, wage, minutes, changed, memo)
                result.add(item)
            }
            close()
        }
        return result
    }

    fun createWorkItem(wage: Int, minutes: Int, changed: String, memo: String) {
        val sql = "INSERT INTO salaries VALUES (?, ?, ?, ?)"
        val bindArgs = arrayOf(
                wage.toString(),
                minutes.toString(),
                changed,
                memo
        )
        db.compileStatement(sql).apply {
            bindAllArgsAsStrings(bindArgs)
            execute()
            close()
        }
    }

    fun updateWorkTime(targetRowId: Int, wage: Int, minutes: Int, changed: String, memo: String) {
        val sql = "UPDATE salaries SET wage = ?, minutes = ?, changed = ?, memo = ? WHERE ROWID = ?"
        val bindArgs = arrayOf(
                wage.toString(),
                minutes.toString(),
                changed,
                memo,
                targetRowId.toString()
        )
        db.compileStatement(sql).apply {
            bindAllArgsAsStrings(bindArgs)
            execute()
            close()
        }
    }

    fun deleteWorkItem(targetRowId: Int) {
        val sql = "DELETE FROM salaries WHERE ROWID = $targetRowId"
        db.execSQL(sql)
    }

    fun close() {
        db.close()
    }

}